docs: add Remix integration page (#1281)
vanilla-extract-css/vanilla-extract: site/docs/integrations/remix.md

---
title: Remix
parent: integrations
---

# Remix

> A plugin for integrating vanilla-extract with [Remix](https://remix.run).
> 
> See [Vanilla Extract | Remix](https://remix.run/docs/en/main/styling/vanilla-extract) for documentation.
> 
> Remix's Vanilla Extract integration is Vite-based.
> See also [Vanilla Extract's Vite integration](/vite).

Reviewer: ```suggestion
[Remix](https://remix.run) provides support for Vanilla Extract out of the box. See [Vanilla Extract | Remix](https://remix.run/docs/en/main/styling/vanilla-extract) for details.

Remix's (unstable) Vite compiler also works with the [Vite integration](/vite). It's as simple as adding the `@vanilla-extract/vite-plugin` to your Vite config:

```js
import { unstable_vitePlugin as remix } from '@remix-run/dev';
import { vanillaExtractPlugin } from '@vanilla-extract/vite-plugin';
import { defineConfig } from 'vite';

export default defineConfig({
  plugins: [remix(), vanillaExtractPlugin()]
});
```

See [Vite (Unstable) | Remix](https://remix.run/docs/en/main/future/vite#add-vanilla-extract-plugin) for more details.
```

---
title: Remix
parent: integrations
---

# Remix

> [Remix](https://remix.run) provides support for Vanilla Extract out of the box. See [Vanilla Extract | Remix](https://remix.run/docs/en/main/styling/vanilla-extract) for details.
> 
> Remix's (unstable) Vite compiler also works with the [Vite integration](/vite). It's as simple as adding the `@vanilla-extract/vite-plugin` to your Vite config:
> 
> ```js
> import { unstable_vitePlugin as remix } from '@remix-run/dev';
> import { vanillaExtractPlugin } from '@vanilla-extract/vite-plugin';
> import { defineConfig } from 'vite';
> 
> export default defineConfig({
>   plugins: [remix(), vanillaExtractPlugin()]
> });
> ```
> 
> See [Vite (Unstable) | Remix](https://remix.run/docs/en/main/future/vite#add-vanilla-extract-plugin) for more details.

Reviewer: ````suggestion
[Remix](https://remix.run) provides support for Vanilla Extract out of the box. See [Vanilla Extract | Remix](https://remix.run/docs/en/main/styling/vanilla-extract) for details.

Remix's (unstable) Vite compiler works with the [Vite integration]. It's as simple as adding the `@vanilla-extract/vite-plugin` to your Vite config:

```js
import { unstable_vitePlugin as remix } from '@remix-run/dev';
import { vanillaExtractPlugin } from '@vanilla-extract/vite-plugin';
import { defineConfig } from 'vite';

export default defineConfig({
  plugins: [remix(), vanillaExtractPlugin()]
});
```

See [Vite (Unstable) | Remix](https://remix.run/docs/en/main/future/vite#add-vanilla-extract-plugin) for more details.

[Vite integration]: /documentation/integrations/vite
````

---
title: Remix
parent: integrations
---

# Remix

[Remix](https://remix.run) provides support for Vanilla Extract out of the box. See [Vanilla Extract | Remix](https://remix.run/docs/en/main/styling/vanilla-extract) for details.

Remix's (unstable) Vite compiler works with the [Vite integration]. It's as simple as adding the `@vanilla-extract/vite-plugin` to your Vite config:

```js
import { unstable_vitePlugin as remix } from '@remix-run/dev';
import { vanillaExtractPlugin } from '@vanilla-extract/vite-plugin';
import { defineConfig } from 'vite';

export default defineConfig({
  plugins: [remix(), vanillaExtractPlugin()]
});
```

See [Vite (Unstable) | Remix](https://remix.run/docs/en/main/future/vite#add-vanilla-extract-plugin) for more details.

> [Vite integration]: /documentation/integrations/vite

Reviewer: ```suggestion
[vite integration]: /documentation/integrations/vite
```
